src/Storages/StorageFile.cpp

```diff
@@ -691,8 +691,7 @@ Pipe StorageFile::read(
         const auto get_columns_for_format = [&]() -> ColumnsDescription
         {
             if (isColumnOriented())
-                return ColumnsDescription{
-                    storage_snapshot->getSampleBlockForColumns(column_names).getNamesAndTypesList()};
```

Comment:
Bug was here, such `ColumnsDescription` didn't contain any information about default expressions for columns.